app/src/main/java/tv/vizbee/demo/activity/MainActivity.kt

```diff
@@ -47,10 +80,88 @@ class MainActivity : AppCompatActivity(),
             .commit()
     }
 
+    override fun showUserLoginFragment() {
+        val tag = UserLoginFragment::class.java.simpleName
+
+        if (supportFragmentManager.findFragmentByTag(tag) != null) {
+            // Do not add fragment if its already on the stack
+            return
+        }
+
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.container, UserLoginFragment.newInstance(), tag)
+            .addToBackStack(null)
+            .commit()
+    }
+
+    override fun popBackStack() {
+        supportFragmentManager.popBackStack()
+    }
+
     override fun onCreateOptionsMenu(menu: Menu): Boolean {
         // Inflate the menu; this adds items to the action bar if it is present.
         menuInflater.inflate(R.menu.menu_main, menu)
         return true
     }
 
+    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
+        val item = menu.findItem(R.id.account)
+        val context = VizbeeWrapper.context?.get()
+        context?.let {
+            val isUserLoggedIn = (SharedPreferenceHelper.getAuthToken()?.isNotEmpty() == true)
+            item.title =
+                if (isUserLoggedIn) getString(R.string.sign_out) else getString(R.string.sign_in)
+        }
+
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val context = VizbeeWrapper.context?.get()
+        context?.let {
+            val authToken = SharedPreferenceHelper.getAuthToken()
+            val isUserLoggedIn = (authToken?.isNotEmpty() == true)
+            when (item.itemId) {
+                R.id.account -> if (isUserLoggedIn) {
+                    signOut(authToken)
+                } else {
+                    showUserLoginFragment()
+                }
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun signOut(authToken: String?) {
+
+        authToken?.let {
+            val apiInterface = NetworkInstance.getInstance().create(ApiInterface::class.java)
```

Comment:
simple names apInterface doesn’t make much sense. Naming them as LoginAPI or something is better. Check for other sources in FN or somewhere.

Reply:
Changed.